docs/development.md

```diff
@@ -1,16 +1,48 @@
 ## Pre-requisites
 
-You'll have to have [Rust toolchain](https://rustup.rs/) installed as well as LLVM, Clang and CMake in addition to usual developer tooling (Ubuntu example):
+You'll have to have [Rust toolchain](https://rustup.rs/) installed as well as LLVM, Clang and CMake in addition to usual developer tooling.
+
+Below are some examples of how to install these dependencies on different operating systems.
+
+### Ubuntu
+
 ```bash
 sudo apt-get install llvm clang cmake
 ```
 
+### macOS M1
+
+1. Install via Homebrew:
+
+   ```sh
+   $ brew install llvm@15 clang cmake
+   ```
+
+2. Add `llvm` to your `~/.zshrc` or `~/.bashrc`:
+
+   ```sh
+   export PATH="/opt/homebrew/opt/llvm@15/bin:$PATH"
+   ```
+
+3. Activate the changes:
+
+   ```sh
+   $ source ~/.zshrc
+   ```
+
+4. Verify that `llvm` is installed:
+
+   ```sh
+   $ llvm-config --version
+   ```
```

Comment:
This is not specific for M1 in any way, so let's remove M1 mention. Also no need for extra indentation of code snippets and we did use `bash` rather than `sh` in all other examples and without `$ `, so it would be nice to keep it consistent.